src/QsCompiler/CompilationManager/AssemblyLoader.cs

```diff
@@ -172,6 +172,53 @@ public static bool LoadReferencedAssembly(Uri asm, out References.Headers header
             return compilation != null && IsValidCompilation(compilation);
         }
 
+        /// <summary>
+        /// Loads any QIR byte code included as a resource from <paramref name="assemblyFile"/>.
+        /// </summary>
+        /// <param name="assemblyFile">The reader for the byte stream of a dotnet DLL from which to load the compilation.</param>
+        /// <param name="qirByteCode">The QIR byte code included as a resource of <paramref name="assemblyFile"/>.</param>
+        /// <returns>
+        /// True if <paramref name="assemblyFile"/> includes a suitable resource, false otherwise.
+        /// </returns>
+        public static bool LoadQirByteCode(
+            PEReader assemblyFile,
```

Comment:
I think this should be `FileInfo` instead of `PEReader`, and the `PEReader` object should be created inside.

Reply:
Since this is something done in more than one place, it might make sense to create an auxiliary method.

Reply:
Actually, most of the method can be common. I would implement two generic methods:
One to get a `PEReader`
```csharp
private static PEReader GetAssemblyReader(FileInfo assembly);
```
And one to load a resource to a stream:
```csharp
private static void LoadResource(string resourceName, PEReader assemblyReader);
```

Reply:
Hmm, I'm not seeing a `FileInfo` to `PEReader` method being useful to anywhere else in this file. Did you mean somewhere else?

Reply:
`LoadReferencedAssembly` does not have a `FileInfo` argument but it creates a `PEReader` object from a `string` argument. The creation of the `PEReader` would be common both to this method and the `LoadReferencedAssembly` one.